pcdsutils/log.py

```diff
@@ -12,8 +12,11 @@
 import queue as queue_module
 import socket
 import sys
+import traceback
+import types
 import typing
 import warnings
+from typing import Optional, Tuple, Union
```

Comment:
Is there a reason we take out the `typing.` for some of the type hints but not all of them?  Are hints like `TextIO` considered not well-known enough to omit the parent package?

Reply:
A good reason? No. I tend to go back and forth and annoy reviewers with this.

1. I like to import the common ones by name `Tuple`, `Union`, `List` are used everywhere (to reduce typing)
2. Uncommon ones or those that get used perhaps once like `TextIO` I tend to do with the qualified name - i.e., `typing.TextIO`

The stylistic issue of which to choose will go away at some point when we are on 3.9+ everywhere (since we can do `list[int]` instead of `typing.List[int]`)

Reply:
I typically just `from typing import x` for all of them.

In general I'm pretty happy with using from imports for everything except when the context is important to understanding the code. For example, if there was a common namespace collision I'd probably avoid the from import...

It's not very important in most cases I think.

Reply:
I also avoid `from` imports when the function name is super generic... like `post` or `read` or something like that. `modulename.post` is much clearer in the code.